cli/azd/cmd/util.go

```diff
@@ -453,16 +454,23 @@ func promptTemplate(ctx context.Context, message string, askOne Asker) (string,
 		Options: templateNames,
 		Default: templateNames[0],
 	}, &selectedTemplateIndex); err != nil {
-		return "", fmt.Errorf("prompting for template: %w", err)
+		return result, fmt.Errorf("prompting for template: %w", err)
 	}
 
 	if selectedTemplateIndex == 0 {
-		return "", nil
+		return result, nil
 	}
 
-	log.Printf("Selected template: %s", fmt.Sprint(templateNames[selectedTemplateIndex]))
+	selectedTemplateName := templateNames[selectedTemplateIndex]
+	log.Printf("Selected template: %s", fmt.Sprint(selectedTemplateName))
 
-	return templateNames[selectedTemplateIndex], nil
+	for _, template := range templates {
```

Comment:
Isn't this always going to be `templates[selectedTemplateIndex - 1]` (`-1` since the first item in the selction list is the "empty template" thing we add?

Reply:
That might be right now, but there's nothing to ensure the order will always be kept when getting the `templateNames` out of the `templates`.  I would be worried that some change in the future might break the ordering and ended up returning a different template than the selected one.

I see a potential collision issue, however, in my implementation, as we don't have anything to make names unique, other than our fine judgement :)